### ISSUE TITLE
How do I publish into a subfolder of PublishDir?

### ISSUE THREAD

**Issue by ferenc-a** [nulastudio/NetBeauty2#73]
I have 3 executables I publish as a set, and I want them to use a single dotnet runtime.
I tried using NetBeauty2, and worked based on [SharedRuntimeTest](https://github.com/nulastudio/NetBeauty2/tree/master/NetBeautyTest/SharedRuntimeTest).

This is the directory structure in the example:
```
publish
├── runtimes
├── app1
│   ├── app1.dll
│   └── ...
├── app1.exe
```

But it's achieved by setting `<PublishDir>../publish/WsClient</PublishDir>`. I want the PublishDir to be still set to publish, not the subfolder, and tell NetBeauty2 to use this folder structure, but I can't find a way. I tried setting BeautyDir, but it does nothing, and the other variables don't affect the exe, or ruin the relative paths when loading the dlls.

**Comment by liesauer** [comment]
no, you can't. lets says you have 2 apps, app1 and app2, by default, the entire project folder structure should be like this:
```
project
│
├──app1
│   ├── bin
│   │   └── Relase/blah blah/publish
│   │                            ├── app1.exe
│   │                            ├── app1.dll
│   │                            └── ...
│   ├── app1.csproj
│   └── ...
├──app2
│   ├── bin
│   │   └── Relase/blah blah/publish
│   │                            ├── app2.exe
│   │                            ├── app2.dll
│   │                            └── ...
│   ├── app2.csproj
│   └── ...
```

to make them publishing as an application set, this two projects must have the same publish dir, you can publish app1 into app2's `PublishDir` or publish app2 into app1's `PublishDir` or just simply change the `PublishDir` to `${projectFolder}/publish`.

**Comment by ferenc-a** [comment]
I can copy the other app in a separate step, that's not my problem. Even with a single app, this causes problems for me.
There are other files and paths that are dependent on the publish dir, and this would, for example make a db file that was included in the publishdir root move to the app subdir. I don't want to change all paths related to publishdir, just for the sake of NetBeauty2.
What does BeautyDir mean then, if not this? With so many config options related to this already (BeautyDir, BeautyAppHostDir, BeautyAppHostEntry, BeautyLibsDir), shouldn't NetBeauty2 be able to handle this?

**Comment by ferenc-a** [comment]
@liesauer could you please respond to the questions in the last part? What does BeautyDir mean? The documentation is really thin on this.

**Comment by liesauer** [comment]
BeautyDir is where your app root dir is(where the deps.json is)
